libs/common/src/key-management/master-password/abstractions/master-password.service.abstraction.ts

```diff
@@ -44,33 +55,68 @@ export abstract class MasterPasswordServiceAbstraction {
     userId: string,
     userKey?: EncString,
   ) => Promise<UserKey | null>;
+
+  /**
+   * Makes the authentication hash for authenticating to the server with the master password.
+   */
+  abstract makeMasterPasswordAuthenticationData: (
+    password: string,
+    kdf: KdfConfig,
+    salt: MasterPasswordSalt,
+    userId: UserId,
+  ) => Promise<MasterPasswordAuthenticationData>;
+
+  /**
+   * Creates a MasterPasswordUnlockData bundle that encrypts the user-key with a key derived from the password. The
+   * bundle also contains the KDF settings and salt used to derive the key, which are required to decrypt the user-key later.
+   */
+  abstract makeMasterPasswordUnlockData: (
+    password: string,
+    kdf: KdfConfig,
+    salt: MasterPasswordSalt,
+    userKey: UserKey,
+  ) => Promise<MasterPasswordUnlockData>;
+
+  /**
+   * Unwraps a user-key that was wrapped with a password provided KDF settings. The same KDF settings and salt must be provided to unwrap the user-key, otherwise it will fail to decrypt.
+   * @throws If the encryption type is not supported.
+   * @throws If the password, KDF, or salt don't match the original wrapping parameters.
+   */
+  abstract unwrapUserKeyFromMasterPasswordUnlockData: (
+    password: string,
+    masterPasswordUnlockData: MasterPasswordUnlockData,
+  ) => Promise<UserKey>;
 }
 
 export abstract class InternalMasterPasswordServiceAbstraction extends MasterPasswordServiceAbstraction {
   /**
    * Set the master key for the user.
    * Note: Use {@link clearMasterKey} to clear the master key.
+   * @deprecated
    * @param masterKey The master key.
    * @param userId The user ID.
    * @throws If the user ID or master key is missing.
    */
   abstract setMasterKey: (masterKey: MasterKey, userId: UserId) => Promise<void>;
   /**
    * Clear the master key for the user.
+   * @deprecated
```

Comment:
```suggestion
   * @deprecated Interacting with the master-key directly is deprecated.
```

It would be nice to have at least a note on these `InternalMasterPasswordService` deprecations.